Sources_v3/Utils/Internet Connection/Reachability.swift

```diff
@@ -0,0 +1,391 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import SystemConfiguration
+
+enum ReachabilityError: Error {
+    case failedToCreateWithAddress(sockaddr, Int32)
+    case failedToCreateWithHostname(String, Int32)
+    case unableToSetCallback(Int32)
+    case unableToSetDispatchQueue(Int32)
+    case unableToGetFlags(Int32)
+}
+
+extension Notification.Name {
+    static let reachabilityChanged = Notification.Name("io.getstream.StreamChatClient.reachabilityChanged")
+}
+
+class Reachability {
```

Comment:
It'd be great if you cleaned up the implementation here a bit. For example, we don't need the notification center here, we don't need the deprecated methods etc. Since this will be wrapped with our custom class, all of this can go.